rust/src/applayer.rs

```diff
@@ -193,6 +223,48 @@ impl From<i32> for AppLayerResult {
     }
 }
 
+pub enum Frame {}
```

Comment:
I'd like to see the implementation of Frame more Rust like,if there isn't anything I'm aware of preventing it.. I quickly did this impl: https://gist.github.com/jasonish/ae033575f271f72d4d8ab36c400f340d

Ideally something like this before too many parsers are converted?

Reply:
yeah happy to take this route

wondering still about how to deal with the `CFrame` pointer. It makes it more efficient to do something like:

* create frame
* set tx_id
* set event

As the pointer can be used directly. But after the current invocation of the parser is complete the pointer becomes invalid, so holding on to it should not be done. Then only the ID should be used. I guess the only safe way to do this is to not expose the pointer at all...

This is not different in the PR btw.

Reply:
I believe you were asking about some datatype a while ago that would make it impossible to hold onto? Makes sense here, but I don't think its possible.

Reply:
I think in the normal case we'll have a small number of frames, so looking up by id will be a minor cost. But if we have a larger number, say a 100 or so, we'd want to avoid iterating all of them for simple operations.

Reply:
could we somehow force a lifetime requirement on a ptr that can't be satisfied?

Reply:
Maybe.. Will need to wrap the raw pointer in a Rust struct tho that wraps the raw pointer and includes some phantom data, as raw pointers don't have a life time.